src/core/src/op/read_value.cpp

```diff
@@ -151,9 +150,22 @@ bool ReadValue::evaluate(TensorVector& outputs,
     const auto& var_value = variable_values.find(m_variable);
 
     const auto use_context = var_value != variable_values.end() && !var_value->second->get_reset();
-
     auto& output = outputs[0];
-    const auto& input = use_context ? var_value->second->get_state() : inputs[0];
+    Tensor input;
+    if (use_context) {
+        input = var_value->second->get_state();
+    } else {
+        if (!inputs.empty()) {
+            input = inputs[0];
+        } else {
+            auto var_info = m_variable->get_info();
+            OPENVINO_ASSERT(var_info.data_shape.is_static() && var_info.data_type.is_static());
+            const auto& shape = var_info.data_shape.get_shape();
+            const auto& type = var_info.data_type;
+            input = ov::Tensor(type, shape);
+            memset(input.data(), 0, type.size() * shape_size(shape));
```

Comment:
It's unsafe for <8bit types.
```suggestion
            memset(input.data(), 0, input.get_byte_size());
```

Reply:
good catch! thanks
fixed